Build Timeouts on integration tests

### Describe the bug

Builds for PRs fail with a timeout after 6 hours. 

See https://github.com/keycloak/keycloak/actions/runs/3284539014/jobs/5413063587 for an example.

### Version

nightly

### Expected behavior

Builds should complete within ~60 min, 90 minutes max.

### Actual behavior

Builds time out after 6 hours

### How to Reproduce?

This is happening only sometimes, needs to be investigated.

### Anything else?

_No response_

The last line in the logs for

https://github.com/keycloak/keycloak/actions/runs/3284539014/jobs/5413063587 for quarkus-map-hot-rod group 2

``` 
[INFO] Running org.keycloak.testsuite.broker.KcOidcBrokerTest
Oct 20, 2022 12:09:02 AM org.jboss.arquillian.drone.webdriver.factory.HtmlUnitDriverFactory setClientOptions
INFO: Setting HtmlDriver web client options: {htmlunitCssEnabled=false, htmlunitHistoryPageCacheLimit=1}
```

https://github.com/keycloak/keycloak/actions/runs/3294658526/jobs/5443664008 for quarkus-map group 2

```
[INFO] Running org.keycloak.testsuite.broker.KcOidcBrokerTest
Oct 21, 2022 7:32:26 PM org.jboss.arquillian.drone.webdriver.factory.HtmlUnitDriverFactory setClientOptions
INFO: Setting HtmlDriver web client options: {htmlunitCssEnabled=false, htmlunitHistoryPageCacheLimit=1}
```

https://github.com/keycloak/keycloak/actions/runs/3303725600/jobs/5452084697 for quarkus-map group 1

```
[INFO] Running org.keycloak.testsuite.sessionlimits.KcOidcUserSessionLimitsBrokerTest
Oct 22, 2022 4:31:57 PM org.jboss.arquillian.drone.webdriver.factory.HtmlUnitDriverFactory setClientOptions
INFO: Setting HtmlDriver web client options: {htmlunitCssEnabled=false, htmlunitHistoryPageCacheLimit=1}
```

https://github.com/keycloak/keycloak/actions/runs/3309185559/jobs/5462178091 Model Tests

```
======== Start of Profile legacy-jpa+infinispan-offline-sessions-preloading
....
2022-10-24T01:04:44.7125945Z 01:04:44,688 INFO  [org.keycloak.testsuite.model.session.OfflineSessionPersistenceTest] (main) testLazyOfflineUserSessionFetching STARTED
2022-10-24T01:04:45.1135398Z 01:04:44,970 DEBUG [org.keycloak.testsuite.model.KeycloakModelParameters] (main) Closing KeycloakSessionFactory 21 (from Time-limited test thread)
2022-10-24T01:04:45.1136733Z 01:04:44,978 INFO  [org.hibernate.orm.connections.pooling] (main) HHH10001008: Cleaning up connection pool [jdbc:h2:mem:test;DB_CLOSE_DELAY=-1;NON_KEYWORDS=VALUE]
2022-10-24T01:04:45.1137649Z 01:04:44,978 DEBUG [org.keycloak.testsuite.model.KeycloakModelParameters] (main) Creating factory 26 in main using the following configuration:
...
2022-10-24T01:04:48.2115719Z 01:04:48,106 ERROR [org.keycloak.models.sessions.infinispan.initializer.InfinispanCacheInitializer] (main) ExecutionException when computed future. Errors: 20
2022-10-24T01:04:48.2116088Z java.util.concurrent.ExecutionException: org.infinispan.commons.CacheException: org.keycloak.models.ModelException: java.lang.NullPointerException
2022-10-24T01:04:48.2116344Z    at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:395)
2022-10-24T01:04:48.2116582Z    at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1999)
2022-10-24T01:04:48.2117055Z    at org.keycloak.models.sessions.infinispan.initializer.InfinispanCacheInitializer.startLoadingImpl(InfinispanCacheInitializer.java:177)
2022-10-24T01:04:48.2117503Z    at org.keycloak.models.sessions.infinispan.initializer.InfinispanCacheInitializer.startLoading(InfinispanCacheInitializer.java:111)
2022-10-24T01:04:48.2117959Z    at org.keycloak.models.sessions.infinispan.initializer.DBLockBasedCacheInitializer.startLoading(DBLockBasedCacheInitializer.java:75)
2022-10-24T01:04:48.2118333Z    at org.keycloak.models.sessions.infinispan.initializer.CacheInitializer.loadSessions(CacheInitializer.java:42)
2022-10-24T01:04:48.2118747Z    at org.keycloak.models.sessions.infinispan.InfinispanUserSessionProviderFactory$2.run(InfinispanUserSessionProviderFactory.java:191)
2022-10-24T01:04:48.2119108Z    at org.keycloak.models.utils.KeycloakModelUtils.runJobInTransaction(KeycloakModelUtils.java:260)
2022-10-24T01:04:48.2119640Z    at org.keycloak.models.sessions.infinispan.InfinispanUserSessionProviderFactory.loadPersistentSessions(InfinispanUserSessionProviderFactory.java:172)
2022-10-24T01:04:48.2120071Z    at org.keycloak.models.sessions.infinispan.InfinispanUserSessionProviderFactory$1.lambda$onEvent$0(InfinispanUserSessionProviderFactory.java:127)
2022-10-24T01:04:48.2120376Z    at org.keycloak.models.utils.KeycloakModelUtils.runJobInTransaction(KeycloakModelUtils.java:260)
2022-10-24T01:04:48.2120733Z    at org.keycloak.models.utils.KeycloakModelUtils.runJobInTransactionWithTimeout(KeycloakModelUtils.java:367)
2022-10-24T01:04:48.2121183Z    at org.keycloak.models.sessions.infinispan.InfinispanUserSessionProviderFactory$1.onEvent(InfinispanUserSessionProviderFactory.java:123)
2022-10-24T01:04:48.2121503Z    at org.keycloak.services.DefaultKeycloakSessionFactory.publish(DefaultKeycloakSessionFactory.java:92)
2022-10-24T01:04:48.2121862Z    at org.keycloak.testsuite.model.KeycloakModelTest.createKeycloakSessionFactory(KeycloakModelTest.java:318)
2022-10-24T01:04:48.2122242Z    at org.keycloak.testsuite.model.KeycloakModelTest.reinitializeKeycloakSessionFactory(KeycloakModelTest.java:327)
2022-10-24T01:04:48.2122772Z    at org.keycloak.testsuite.model.session.OfflineSessionPersistenceTest.testPersistenceSingleNodeDeleteRealm(OfflineSessionPersistenceTest.java:119)
2022-10-24T01:04:48.2123020Z    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2022-10-24T01:04:48.2123308Z    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2022-10-24T01:04:48.2123673Z    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-10-24T01:04:48.2123855Z    at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2022-10-24T01:04:48.2124096Z    at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
2022-10-24T01:04:48.2124361Z    at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
2022-10-24T01:04:48.2124629Z    at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
2022-10-24T01:04:48.2124895Z    at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
2022-10-24T01:04:48.2125146Z    at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
2022-10-24T01:04:48.2125393Z    at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
2022-10-24T01:04:48.2125568Z    at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
2022-10-24T01:04:48.2125762Z    at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
2022-10-24T01:04:48.2125999Z    at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
2022-10-24T01:04:48.2126181Z    at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
2022-10-24T01:04:48.2126436Z    at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
2022-10-24T01:04:48.2126686Z    at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
2022-10-24T01:04:48.2126867Z    at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
2022-10-24T01:04:48.2127055Z    at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
2022-10-24T01:04:48.2127267Z    at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
2022-10-24T01:04:48.2127459Z    at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
2022-10-24T01:04:48.2127648Z    at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
2022-10-24T01:04:48.2127904Z    at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
2022-10-24T01:04:48.2128075Z    at org.junit.rules.RunRules.evaluate(RunRules.java:20)
2022-10-24T01:04:48.2128262Z    at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
2022-10-24T01:04:48.2128497Z    at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
2022-10-24T01:04:48.2128755Z    at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:377)
2022-10-24T01:04:48.2129046Z    at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:284)
2022-10-24T01:04:48.2129325Z    at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:248)
2022-10-24T01:04:48.2129574Z    at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:167)
2022-10-24T01:04:48.2129862Z    at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:456)
2022-10-24T01:04:48.2130112Z    at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:169)
2022-10-24T01:04:48.2130341Z    at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:595)
2022-10-24T01:04:48.2130573Z    at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:581)
2022-10-24T01:04:48.2130879Z Caused by: org.infinispan.commons.CacheException: org.keycloak.models.ModelException: java.lang.NullPointerException
2022-10-24T01:04:48.2131327Z    at org.keycloak.models.sessions.infinispan.initializer.InfinispanCacheInitializer.lambda$startLoadingImpl$0(InfinispanCacheInitializer.java:164)
2022-10-24T01:04:48.2131630Z    at org.infinispan.manager.impl.LocalClusterExecutor.lambda$submitConsumer$3(LocalClusterExecutor.java:78)
2022-10-24T01:04:48.2131909Z    at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859)
2022-10-24T01:04:48.2132179Z    at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:837)
2022-10-24T01:04:48.2132491Z    at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
2022-10-24T01:04:48.2132799Z    at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2088)
2022-10-24T01:04:48.2133097Z    at org.infinispan.manager.impl.LocalClusterExecutor.lambda$localInvocation$6(LocalClusterExecutor.java:97)
...
2022-10-24T01:04:48.2142651Z Caused by: java.lang.NullPointerException
2022-10-24T01:04:48.2143038Z    at org.hibernate.loader.plan.exec.process.internal.AbstractRowReader.resolveEntityKey(AbstractRowReader.java:104)
2022-10-24T01:04:48.2143417Z    at org.hibernate.loader.plan.exec.process.internal.AbstractRowReader.resolveEntityKey(AbstractRowReader.java:120)
2022-10-24T01:04:48.2143799Z    at org.hibernate.loader.plan.exec.process.internal.AbstractRowReader.resolveEntityKey(AbstractRowReader.java:108)
2022-10-24T01:04:48.2144137Z    at org.hibernate.loader.plan.exec.process.internal.AbstractRowReader.readRow(AbstractRowReader.java:71)
2022-10-24T01:04:48.2144500Z    at org.hibernate.loader.plan.exec.internal.EntityLoadQueryDetails$EntityLoaderRowReader.readRow(EntityLoadQueryDetails.java:288)
2022-10-24T01:04:48.2144900Z    at org.hibernate.loader.plan.exec.process.internal.ResultSetProcessorImpl.extractRows(ResultSetProcessorImpl.java:157)
2022-10-24T01:04:48.2145308Z    at org.hibernate.loader.plan.exec.process.internal.ResultSetProcessorImpl.extractResults(ResultSetProcessorImpl.java:94)
2022-10-24T01:04:48.2145707Z    at org.hibernate.loader.plan.exec.internal.AbstractLoadPlanBasedLoader.executeLoad(AbstractLoadPlanBasedLoader.java:105)
2022-10-24T01:04:48.2146090Z    at org.hibernate.loader.entity.plan.AbstractLoadPlanBasedEntityLoader.load(AbstractLoadPlanBasedEntityLoader.java:285)
2022-10-24T01:04:48.2146400Z    at org.hibernate.persister.entity.AbstractEntityPersister.doLoad(AbstractEntityPersister.java:4521)
2022-10-24T01:04:48.2146703Z    at org.hibernate.persister.entity.AbstractEntityPersister.load(AbstractEntityPersister.java:4511)
2022-10-24T01:04:48.2147062Z    at org.hibernate.event.internal.DefaultLoadEventListener.loadFromDatasource(DefaultLoadEventListener.java:571)
2022-10-24T01:04:48.2147369Z    at org.hibernate.event.internal.DefaultLoadEventListener.doLoad(DefaultLoadEventListener.java:539)
2022-10-24T01:04:48.2147663Z    at org.hibernate.event.internal.DefaultLoadEventListener.load(DefaultLoadEventListener.java:208)
2022-10-24T01:04:48.2148042Z    at org.hibernate.event.internal.DefaultLoadEventListener.proxyOrLoad(DefaultLoadEventListener.java:327)
2022-10-24T01:04:48.2148355Z    at org.hibernate.event.internal.DefaultLoadEventListener.doOnLoad(DefaultLoadEventListener.java:108)
2022-10-24T01:04:48.2148658Z    at org.hibernate.event.internal.DefaultLoadEventListener.onLoad(DefaultLoadEventListener.java:74)
2022-10-24T01:04:48.2149062Z    at org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:118)
2022-10-24T01:04:48.2149301Z    at org.hibernate.internal.SessionImpl.fireLoadNoChecks(SessionImpl.java:1231)
2022-10-24T01:04:48.2149513Z    at org.hibernate.internal.SessionImpl.fireLoad(SessionImpl.java:1220)
2022-10-24T01:04:48.2149724Z    at org.hibernate.internal.SessionImpl.access$2100(SessionImpl.java:202)
2022-10-24T01:04:48.2149977Z    at org.hibernate.internal.SessionImpl$IdentifierLoadAccessImpl.doLoad(SessionImpl.java:2835)
2022-10-24T01:04:48.2150237Z    at org.hibernate.internal.SessionImpl$IdentifierLoadAccessImpl.lambda$load$1(SessionImpl.java:2812)
2022-10-24T01:04:48.2150493Z    at org.hibernate.internal.SessionImpl$IdentifierLoadAccessImpl.perform(SessionImpl.java:2768)
2022-10-24T01:04:48.2150859Z    at org.hibernate.internal.SessionImpl$IdentifierLoadAccessImpl.load(SessionImpl.java:2812)
2022-10-24T01:04:48.2151057Z    at org.hibernate.internal.SessionImpl.find(SessionImpl.java:3400)
2022-10-24T01:04:48.2151252Z    at org.hibernate.internal.SessionImpl.find(SessionImpl.java:3362)
2022-10-24T01:04:48.2151487Z    at jdk.internal.reflect.GeneratedMethodAccessor279.invoke(Unknown Source)
2022-10-24T01:04:48.2151806Z    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-10-24T01:04:48.2152042Z    at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2022-10-24T01:04:48.2152395Z    at org.keycloak.connections.jpa.PersistenceExceptionConverter.invoke(PersistenceExceptionConverter.java:60)
2022-10-24T01:04:48.2152477Z    ... 28 more
...
2022-10-24T01:04:48.2235170Z 01:04:48,130 DEBUG [org.keycloak.services.DefaultKeycloakSessionFactory] (main) Set default provider for client to jpa
2022-10-24T01:04:48.2235412Z 01:04:48,134 WARN  [org.jgroups.stack.Configurator] (main) JGRP000014: STABLE.stability_delay has been deprecated: always 0
2022-10-24T01:04:48.2235865Z 01:04:48,136 INFO  [org.keycloak.connections.infinispan.InfinispanUtil] (main) Configured jgroups transport with the channel name: node-28
2022-10-24T01:04:48.2236242Z 01:04:48,143 WARN  [org.infinispan.PERSISTENCE] (main) ISPN000554: jboss-marshalling is deprecated and planned for removal
2022-10-24T01:04:48.2236732Z 01:04:48,148 INFO  [org.infinispan.CONTAINER] (main) ISPN000556: Starting user marshaller 'org.infinispan.jboss.marshalling.core.JBossUserMarshaller'
2022-10-24T01:04:48.2237076Z 01:04:48,163 WARN  [org.jgroups.protocols.UDP] (main) JGRP000015: the send buffer of socket MulticastSocket was set to 1.00MB, but the OS only allocated 212.99KB
2022-10-24T01:04:48.2237356Z 01:04:48,163 WARN  [org.jgroups.protocols.UDP] (main) JGRP000015: the receive buffer of socket MulticastSocket was set to 20.00MB, but the OS only allocated 212.99KB
2022-10-24T01:04:48.2237629Z 01:04:48,163 WARN  [org.jgroups.protocols.UDP] (main) JGRP000015: the send buffer of socket MulticastSocket was set to 1.00MB, but the OS only allocated 212.99KB
2022-10-24T01:04:48.2237906Z 01:04:48,163 WARN  [org.jgroups.protocols.UDP] (main) JGRP000015: the receive buffer of socket MulticastSocket was set to 25.00MB, but the OS only allocated 212.99KB
2022-10-24T01:04:48.4195364Z 01:04:48,350 INFO  [org.keycloak.connections.infinispan.DefaultInfinispanConnectionProviderFactory] (main) Node name: node-28, Site name: null
2022-10-24T06:47:18.5297792Z ##[error]The operation was canceled.
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -58,6 +58,7 @@ jobs:
     name: Unit Tests
     runs-on: ubuntu-latest
     needs: build
+    timeout-minutes: 20
     steps:
       - uses: actions/checkout@v3
       - uses: actions/setup-java@v3
@@ -104,6 +105,7 @@ jobs:
     name: Crypto Tests
     runs-on: ubuntu-latest
     needs: build
+    timeout-minutes: 20
     steps:
       - uses: actions/checkout@v3
       - uses: actions/setup-java@v3
@@ -148,6 +150,7 @@ jobs:
     name: Model Tests
     runs-on: ubuntu-latest
     needs: build
+    timeout-minutes: 60
     steps:
       - uses: actions/checkout@v3
       - uses: actions/setup-java@v3
@@ -194,6 +197,7 @@ jobs:
     name: Base testsuite
     needs: build
     runs-on: ubuntu-latest
+    timeout-minutes: 70
     strategy:
       matrix:
         server: ['quarkus', 'quarkus-map', 'quarkus-map-hot-rod']
@@ -278,6 +282,7 @@ jobs:
     name: Base testsuite (fips)
     needs: build
     runs-on: ubuntu-latest
+    timeout-minutes: 30
     strategy:
       matrix:
         server: ['bcfips-nonapproved-pkcs12']
@@ -350,6 +355,7 @@ jobs:
     name: Base testsuite (postgres)
     needs: build
     runs-on: ubuntu-latest
+    timeout-minutes: 80
     strategy:
       matrix:
         server: ['undertow-map-jpa']
@@ -448,6 +454,7 @@ jobs:
     name: Quarkus Test Clustering
     needs: build
     runs-on: ubuntu-latest
+    timeout-minutes: 35
     env:
       MAVEN_OPTS: -Xmx1024m
     steps:
@@ -508,6 +515,7 @@ jobs:
     name: Quarkus Tests
     needs: build
     runs-on: ubuntu-latest
+    timeout-minutes: 100
     env:
       MAVEN_OPTS: -Xmx1024m
     steps:

